### DISCUSSION TITLE
Question about retrieving TaskScheduled EventType Input in orchestration instance

### DISCUSSION THREAD

**Discussion by mengschin** [Azure/azure-functions-durable-extension#1759] [Q&A] [answered]
Hello, 

While performing IDurableOrchestrationClient's GetStatusAsync, by marking showInput, showHistoryOutput and showHistory as true. I'm not able to see the `TaskScheduled EventType` from the history retrieved. On the other hand, Input is always stored as _null_ in History table for `TaskScheduled EventType`

![image](https://user-images.githubusercontent.com/59546764/82636294-dd1b2e80-9c34-11ea-9aea-b938b3fbc2bd.png)
* `TaskScheduled EventType` is not included.

Is there a way we can get the Input the particular activity/task received?

Thanks.

**Comment by mengschin** [comment]
Hi @cgillum , I`m not sure if this question is being answered previously. 
Your comments would be greatly appreciated.

**Comment by cgillum** [comment, marked as answer]
> I'm not able to see the TaskScheduled EventType from the history retrieved

We have logic that actually [removes the TaskScheduled events after the tasks have completed](https://github.com/Azure/azure-functions-durable-extension/blob/6eeedbf0889b89c898514e8063d491248bb2a903/src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs#L740-L755) to keep the size of the history smaller. Part of me regrets this decision because there is less visibility available to you, though I fear removing it would be a breaking change for some. It's something I'm considering revisiting either as a breaking change update or as a feature of alternate Durable Functions backends.

> Is there a way we can get the Input the particular activity/task received?

This is not possible currently. Part of the reason is the answer above. However, the other part of the reason is that the Durable Task Framework's Azure Storage backend does not store this information in Azure Storage. It wasn't strictly required to be in the history table since it already exists in the queue message and wasn't needed for replay to work.
